lib/config/ChainConfigManager.ts

```diff
@@ -0,0 +1,300 @@
+import { ChainId } from '@uniswap/sdk-core';
+import { RoutingType } from '../constants';
+
+type CommonConfig = {
+  routingType: RoutingType;
+  priceImprovementBps?: number;
+  stdAuctionPeriodSecs?: number;
+  deadlineBufferSecs?: number;
```

Comment:
Instead of having optionals here, could we strictly type them for the config types where they exist? ie these ones should exist on dutch, relay, and dutch_v2 but never for classic

Reply:
You're right that these can be moved to the `DutchConfig`. They still need to be optional since they act as overrides to the default value.